File: yangxj96-starter/yangxj96-starter-db/src/main/java/com/yangxj96/saas/starter/db/annotation/EnableDynamicDatasource.java
package com.yangxj96.saas.starter.db.annotation;

import com.yangxj96.saas.starter.db.autoconfigure.DynamicDatasourceAutoConfiguration;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 开启动态数据源
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Configuration
@Import(DynamicDatasourceAutoConfiguration.class)
public @interface EnableDynamicDatasource {

}

File: yangxj96-starter/yangxj96-starter-db/src/main/java/com/yangxj96/saas/starter/db/annotation/EnableDynamicDatasource.kt
/*
 * Copyright (c) 2018 - 2023
 * 作者：杨新杰(Jack Young)
 * 邮箱：yangxj96@gmail.com
 * 博客：www.yangxj96.com
 * 日期：2023-09-13 00:04:11
 * Copyright (c) 2018 - 2023
 */

package com.yangxj96.saas.starter.db.annotation

import com.yangxj96.saas.starter.db.autoconfigure.DynamicDatasourceAutoConfiguration
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import

/**
 * 动态数据源注解,主要用于确定注入动态数据源
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@Configuration
@Import(DynamicDatasourceAutoConfiguration::class)
annotation class EnableDynamicDatasource
